app/controllers/shows_controller.rb

```diff
@@ -4,22 +4,16 @@ class ShowsController < ApplicationController
 
   def index
     if params[:filter]
-      @shows = Show.all
-      @start_date = params[:filter][:start_date].to_datetime
-      @end_date = params[:filter][:end_date].to_datetime
-
-      if params[:filter][:start_date].present? && params[:filter][:end_date].present?
-        if @start_date == @end_date
-          @shows = Performance.shows_on(@start_date)
-        elsif @start_date > @end_date
-          @shows = Show.all
+      if params[:filter][:start_date].present? || params[:filter][:end_date].present?
+        if params[:filter][:start_date].present? && params[:filter][:end_date].present? && params[:filter][:start_date].to_datetime > params[:filter][:end_date].to_datetime
           flash[:notice] = "Uh oh! Please make sure your start date comes before your end date!"
         else
-          @shows = Performance.shows_between(@start_date, @end_date)
+          @fringe_start_date = params[:filter][:start_date].present? ? params[:filter][:start_date].to_datetime + 0.25 : params[:filter][:end_date].to_datetime + 0.25
```

Comment:
What is this + 0.25 doing?

Reply:
it adds on 1/4 of a day i.e. 6hrs
i considered using Time.new(blah blah blah) to make it clearer and match the @fringe_end_date construction but ended up taking the shorter route